numerics/global_optimization.hpp

```diff
@@ -49,7 +51,10 @@ class MultiLevelSingleLinkage {
     Argument centre;
     std::array<Difference<Argument>, 3> vertices;
 
-    Cube<NormType> Measure() const;
+    // If some of the dimensions of the box are zero, we handle the problem as a
+    // 1- or 2-dimensional problem.  This only affects the computation of rₖ.
+    using Measure = std::variant<NormType, Square<NormType>, Cube<NormType>>;
```

Comment:
Do we really expect to have cases where we do not know the dimension of the problem at runtime?

Reply:
Done.  That's indeed nicer.